sdk/tests/conformance2/uniforms/incompatible-texture-type-for-sampler.html

```diff
@@ -289,19 +296,46 @@
                 const internalFormat = textureInfo.internalFormat;
                 const desc = wtu.glEnumToString(gl, internalFormat);
                 const info = textureInternalFormatInfo[internalFormat];
+
+                // The texture object can have two values of TEXTURE_COMPARE_MODE, NONE or
+                // COMPARE_REF_TO_TEXTURE. However, the sampler can have three states:
+                // No sampler object bound, sampler object with NONE, and sampler object with
+                // COMPARE_REF_TO_TEXTURE. When a sampler object is bound, it overrides the
+                // texture object's state. We test 2*3=6 possible combinations of state.
+
+                // First test the three states that result in TEXTURE_COMPARE_MODE being NONE.
+                let expected = samplerType == info.samplerType ? gl.NONE : gl.INVALID_OPERATION;
                 gl.bindTexture(target, textureInfo.texture);
-                gl.texParameteri(target, gl.TEXTURE_COMPARE_MODE, gl.NONE);
                 gl.drawArrays(gl.POINTS, 0, 1);
-                let expected = samplerType == info.samplerType ? gl.NONE : gl.INVALID_OPERATION;
-                wtu.glErrorShouldBe(gl, expected, `from draw with ${desc}`);
+                wtu.glErrorShouldBe(gl, expected, `${desc} texture state NONE, no sampler object`);
+                gl.samplerParameteri(samplerObject, gl.TEXTURE_COMPARE_MODE, gl.NONE);
+                gl.drawArrays(gl.POINTS, 0, 1);
+                wtu.glErrorShouldBe(gl, expected, `${desc} texture state NONE, sampler state NONE`);
+
+                gl.bindSampler(0, samplerObject);
+                gl.samplerParameteri(samplerObject, gl.TEXTURE_COMPARE_MODE, gl.NONE);
```

Comment:
This looks redundant with the similar call just above the drawArrays call, above.